File: app/Models/Book.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Book extends Model
{
    protected $table = 'book';
    use HasFactory;

    protected $fillable = [
        'title',
        'author',
        'major',
        'description',
        'publish_date',
        'quantity',
        'cover'
    ];

    public $timestamps = false;

    public function loans(): HasMany {
        return $this->hasMany(Loan::class, 'book_id');
    }
}

File: lib/model/book.dart
// ignore_for_file: non_constant_identifier_names, must_be_immutable

class Book {
  int book_id = 0;
  String title = '';
  String author = '';
  String description = '';
  String publish_date = '';
  String major = '';
  int quantity = 0;
  String cover = '';

  Book(this.book_id, this.title, this.author, this.major, this.description,
      this.publish_date, this.quantity, this.cover);
}
